test/coverage/datamodel/UuidValidatorTest.cpp

```diff
@@ -23,16 +26,216 @@ class TestFixture : public testing::Test {
 
 	// Run once per execution of the test application.
 	// Used for setup of all tests. Has access to this instance.
-	TestFixture() = default;
-	~TestFixture() = default;
+	TestUuidValidator() = default;
+	~TestUuidValidator() = default;
 
 	// Run once per execution of the test application.
 	// Used only for global setup outside of tests.
 	static void SetUpTestSuite() {}
 	static void TearDownTestSuite() {}
 };
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+// Helper fn for a fake UUID
+uprotocol::v1::UUID createFakeUuid(uint64_t msb, uint64_t lsb) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(msb);
+	uuid.set_lsb(lsb);
+	return uuid;
+}
+
+// Test valid UUID v8
+TEST_F(TestUuidValidator, ValidUuid) {
+	// Create a valid UUID with version 8 and correct variant (10)
+	uint64_t msb = (8ULL << 12) | (0x123ULL);  // Version 8, counter = 0x123
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	auto [valid, reason] = uprotocol::datamodel::validator::uuid::isUuid(uuid);
+	EXPECT_TRUE(valid);
+	EXPECT_FALSE(reason.has_value());
+}
+
+// Test UUID with wrong version
+TEST_F(TestUuidValidator, WrongVersion) {
+	// Creating a UUID with wrong version (!8)
+	uint64_t msb = (7ULL << 12) | (0x123ULL);  // Version 7, counter = 0x123
+	uint64_t lsb = (2ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 10
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	auto [valid, reason] = uprotocol::datamodel::validator::uuid::isUuid(uuid);
+	EXPECT_FALSE(valid);
+	EXPECT_EQ(reason.value(),
+	          uprotocol::datamodel::validator::uuid::Reason::WRONG_VERSION);
+}
+
+// Test UUID with unsupported variant
+TEST_F(TestUuidValidator, UnsupportedVariant) {
+	// Creating UUID with unsupported variant.(!10)
+	uint64_t msb = (8ULL << 12) | (0x123ULL);  // Version 8, counter = 0x123
+	uint64_t lsb = (3ULL << 62) | (0xFFFFFFFFFFFFULL);  // Variant 11
+
+	uprotocol::v1::UUID uuid = createFakeUuid(msb, lsb);
+
+	auto [valid, reason] = uprotocol::datamodel::validator::uuid::isUuid(uuid);
+	EXPECT_FALSE(valid);
+	EXPECT_EQ(
+	    reason.value(),
+	    uprotocol::datamodel::validator::uuid::Reason::UNSUPPORTED_VARIANT);
+}
+
+// Test UUID from the future
+TEST_F(TestUuidValidator, FromTheFuture) {
+	// Creating UUID with a timestamp in the future
+	uint64_t future_timestamp =
+	    std::chrono::duration_cast<std::chrono::milliseconds>(
+	        std::chrono::system_clock::now().time_since_epoch())
+	        .count() +
+	    100000;  // 100 seconds in the future
```

Comment:
I would recommend doing the time arithmetic on chrono objects before converting via `.count()`
```suggestion
	auto future_time = std::chrono::system_clock::now() + 100s;
	uint64_t future_timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(
	    future_time.time_since_epoch()).count();
```